contracts/ZincAccessor.sol

```diff
@@ -140,18 +110,12 @@ contract ZincAccessor is SignatureValidator, Encoder {
         address _idContract,
         uint256 _purpose,
         address _userAddress,
-        string _message1,
         uint32 _nonce,
-        string _header1,
-        string _header2,
         bytes32 _r,
         bytes32 _s,
         uint8 _v)
     public checknonce(_nonce) returns (bool) {
-        require(checkUserSignature(_userAddress, _message1, _nonce, _header1, _header2, _r, _s, _v));
-        require(
-            keccak256(abi.encodePacked("Remove 0x", encodeAddress(_key), " from 0x", encodeAddress(_idContract), " with purpose ", encodeUInt(_purpose))) ==
-            keccak256(encodeString(_message1)), "Message incorrect");
+        require(verifySignature(_r, _s, _v, REMOVE_ACTION, _nonce, _idContract, _userAddress));
```

Comment:
Hmm, the only problem I see here is that we are not verifying the intended purpose... I think that will be a problem if the user wants to add our Accessor as key management, but nothing stops us from adding it as funds management and stealing all their funds.